tests/nodes/NodeConnectionManager.test.ts

```diff
@@ -0,0 +1,1819 @@
+import type { NodeAddress, NodeData, NodeId, SeedNodes } from '@/nodes/types';
+import type { Host, Port, TLSConfig } from '@/network/types';
+import type { AddressInfo } from 'net';
+import net from 'net';
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+import { DB } from '@matrixai/db';
+import Logger, { LogLevel, StreamHandler } from '@matrixai/logger';
+import { _destroyed } from '@matrixai/async-init/dist/utils';
+import { IdInternal } from '@matrixai/id';
+import PolykeyAgent from '@/PolykeyAgent';
+import KeyManager from '@/keys/KeyManager';
+import NodeGraph from '@/nodes/NodeGraph';
+import NodeConnectionManager from '@/nodes/NodeConnectionManager';
+import ForwardProxy from '@/network/ForwardProxy';
+import ReverseProxy from '@/network/ReverseProxy';
+import GRPCClientAgent from '@/agent/GRPCClientAgent';
+import * as nodesUtils from '@/nodes/utils';
+import * as nodesErrors from '@/nodes/errors';
+import * as keysUtils from '@/keys/utils';
+import * as grpcErrors from '@/grpc/errors';
+import * as grpcUtils from '@/grpc/utils';
+import * as agentErrors from '@/agent/errors';
+import * as nodesPB from '@/proto/js/polykey/v1/nodes/nodes_pb';
+import * as utilsPB from '@/proto/js/polykey/v1/utils/utils_pb';
+import { promise, promisify, sleep, withF } from '@/utils';
+import * as nodesTestUtils from './utils';
+import * as testUtils from '../utils';
+
+describe('NodeConnectionManager test', () => {
+  const logger = new Logger('NodeGraph Test', LogLevel.ERROR, [
+    new StreamHandler(),
+  ]);
+  grpcUtils.setLogger(logger.getChild('grpc'));
+
+  const nodeConnectionManagerLogger = logger.getChild(
+    'nodeConnectionManagerUT',
+  );
```

Comment:
These logger names are incorrect. Also in all other tests, we don't bother with creating child loggers unless actually necessary. We just use the test logger in the first `describe`.

I do use it in network tests because I'm trying to see concurrent logs and see how they do what they're doing. But that I don't think that's what is happening here for `nodeConnectionManagerLogger`.

See `tests/acl/ACL.test.ts` at the beginning:

```
describe(ACL.name, () => {
  const logger = new Logger(`${ACL.name} test`, LogLevel.WARN, [
    new StreamHandler(),
  ]);
```

This probably requires fixing in the other new test files you've created.

Reply:
Also note that `LogLevel` should default to `WARN` in our test modules.

Additionally, there are these log messages occurring that isn't using the new GRPC logger:

```
ERROR:FwdProxy:Failed CONNECT to 128.0.0.1:12345 - ErrorConnectionStart: Send failed (status: -22)
  console.error
    Failed to connect to 128.0.0.1:12345/?nodeId=v4nsohs20h7in6rmhrmok5l4hn1p0sl2lh8fa4d5ab1qclakocqk0 through proxy 127.0.0.1:45499 with status 502

      at Object.<anonymous>.exports.log (node_modules/@grpc/grpc-js/src/logging.ts:57:13)
      at ClientRequest.<anonymous> (node_modules/@grpc/grpc-js/src/http_proxy.ts:269:9)

ERROR:FwdProxy:Failed CONNECT to 127.0.0.1:55047 - ErrorCertChainUnclaimed: Node ID is not claimed by any certificate
  console.error
    Failed to connect to 127.0.0.1:55047/?nodeId=v5niu2rtbmaue1d63ihb43u7tpp4cjdsas7difrrnb1uvhrc3gv10 through proxy 127.0.0.1:38395 with status 526

      at Object.<anonymous>.exports.log (node_modules/@grpc/grpc-js/src/logging.ts:57:13)
      at ClientRequest.<anonymous> (node_modules/@grpc/grpc-js/src/http_proxy.ts:269:9)

ERROR:FwdProxy:Failed CONNECT to 127.0.0.1:35982 - ErrorConnectionStart: Client network socket disconnected before secure TLS connection was established
  console.error
    Failed to connect to 127.0.0.1:35982/?nodeId=vias3m3pv3rcev1a38jlv6j6386p9gqb2us9lhsafqn2tsltr2cm0 through proxy 127.0.0.1:38193 with status 502

      at Object.<anonymous>.exports.log (node_modules/@grpc/grpc-js/src/logging.ts:57:13)
      at ClientRequest.<anonymous> (node_modules/@grpc/grpc-js/src/http_proxy.ts:269:9)

ERROR:fwdProxy:Failed CONNECT to 125.0.0.1:55555 - ErrorConnectionStartTimeout
ERROR:grpc:Failed to connect to 125.0.0.1:55555/?nodeId=vi3et1hrpv2m2lrplcm7cu913kr45v51cak54vm68anlbvuf83ra0 through proxy 127.0.0.1:41575 with status 504
```

Usually any logs coming from `logger` should be above the the test logs shown by `jest`. But this was above `NodeManager.test.ts`.

Reply:
This has been fixed with `NodeConnection.test.ts`.

Reply:
Update the logger names as well and describe names to use the static names like `Class.name` as well @tegefaulkes

Reply:
I've updated the logger names inside the nodes tests.